src/main/java/org/launchcode/water_garden_tour/controllers/TourController.java

```diff
@@ -0,0 +1,70 @@
+package org.launchcode.water_garden_tour.controllers;
+
+
+import org.apache.tomcat.util.http.fileupload.IOUtils;
+import org.launchcode.water_garden_tour.models.data.GardenRepository;
+import org.launchcode.water_garden_tour.models.garden.Garden;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Controller
+@RequestMapping("user")
+public class TourController {
+
+    @Autowired
+    private GardenRepository gardenRepository;
+
+    List<Garden> tourGardens = new ArrayList<>();
+
+    @GetMapping("/gardens/tour/{gardenId}")
+    public String addTourGarden(Model model, @PathVariable("gardenId") Integer gardenId) {
+
+        Optional<Garden> chosenGarden = gardenRepository.findById(gardenId);
+        if (chosenGarden.isPresent()) {
+            Garden tourGarden = chosenGarden.get();
+            if (!tourGardens.contains(tourGarden)) {
+                tourGardens.add(tourGarden);
+            }
+        }
+        model.addAttribute("gardens", gardenRepository.findAll());
```

Comment:
I needed to add the following here to get back to the garden list after adding a garden that had been searched for:
        List<Feature> selectedFeatures = new ArrayList<>();

        model.addAttribute("selectedFeatures", selectedFeatures);
        model.addAttribute("features", featureRepository.findAll());;
        model.addAttribute("searchTerm", "");

Reply:
Cool, I added it.  Didn't notice that your search features had disappeared.  Good catch!